rust/main/agents/relayer/src/msg/op_queue.rs

```diff
@@ -72,27 +72,67 @@ impl OpQueue {
         // The other consideration is whether to put the channel receiver in the OpQueue or in a dedicated task
         // that also holds an Arc to the Mutex. For simplicity, we'll put it in the OpQueue for now.
         let mut message_retry_requests = vec![];
-        while let Ok(message_id) = self.retry_rx.lock().await.try_recv() {
-            message_retry_requests.push(message_id);
+
+        while let Ok(retry_request) = self.retry_receiver.lock().await.try_recv() {
+            let uuid = retry_request.uuid.clone();
+            message_retry_requests.push((
+                retry_request,
+                MessageRetryResponse {
+                    uuid,
+                    evaluated: 0,
+                    matched: 0,
+                },
+            ));
         }
+
         if message_retry_requests.is_empty() {
             return;
         }
+
         let mut queue = self.queue.lock().await;
+        let queue_length = queue.len();
+
         let mut reprioritized_queue: BinaryHeap<_> = queue
             .drain()
             .map(|Reverse(mut op)| {
-                if message_retry_requests.iter().any(|r| r.op_matches(&op)) {
+                let matched_requests: Vec<_> = message_retry_requests
+                    .iter_mut()
+                    .filter_map(|(retry_req, retry_response)| {
+                        // update retry metrics
+                        if retry_req.pattern.op_matches(&op) {
+                            debug!(uuid = retry_req.uuid, "Matched request");
+                            retry_response.matched += 1;
+                            Some(retry_req.uuid.clone())
```

Comment:
seems like this isn't used anymore? think we can go back to using `.any` instead of `filter_map` and avoid this allocation